app/audit/events/QuestionnaireAuditEvent.scala

```diff
@@ -17,13 +17,10 @@
 package audit.events
 
 import models.QuestionnaireModel
-import play.api.libs.json.{JsObject, Json}
-import play.api.mvc.{AnyContent, Request}
-import services.RegistrationAuditEvent
-import uk.gov.hmrc.http.HeaderCarrier
+import play.api.libs.json.Json
 
-class QuestionnaireAuditEvent(questionnaireModel: QuestionnaireModel)(implicit hc:HeaderCarrier, request:Request[AnyContent])
-  extends RegistrationAuditEvent (
-    auditType = "Questionnaire",
-      detail = Json.toJson(questionnaireModel).as[JsObject]
-  )(hc,request)
+case class QuestionnaireAuditEvent(questionnaireModel: QuestionnaireModel)
+
+object QuestionnaireAuditEvent {
+  implicit val format = Json.format[QuestionnaireAuditEvent]
+}
```

Comment:
I think you can delete this Class now actually. Because, the `questionaireModel` should be passed directly to the `auditService.sendEvent`